### PR TITLE
Add resource "aws_iam_policy_attachment" to attach a managed policy to users, roles, or groups

### PR DESCRIPTION
**Pull request by GrayCoder**
This implements #2100 

It follows similar logic as #2273 

It attaches a managed policy to users, roles, and/or groups.

### CODE REVIEW

**Comment by scalp42** [comment]
:+1: thanks a lot @GrayCoder

**Comment by ctiwald** [comment]
:+1:

**Comment by dalehamel** [comment]
@catsby anyone from hashicorp able to review this? This is a feature we really need as well, so we're grabbing this branch to play with.

File: builtin/providers/aws/resource_aws_iam_policy_attachment.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+func resourceAwsIamPolicyAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iamconn
+
+	name := d.Get("name").(string)
+	arn := d.Get("policy_arn").(string)
+	users := expandStringList(d.Get("users").(*schema.Set).List())
+	roles := expandStringList(d.Get("roles").(*schema.Set).List())
+	groups := expandStringList(d.Get("groups").(*schema.Set).List())
+
+	if users == nil && roles == nil && groups == nil {
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_iam_policy_attachment.go]: The method `expandStringList` calls `make([]*string, ...)` and returns the result. Even that slice is empty, it's not `nil`, so this block will never evaluate to `true`, yeah?

File: builtin/providers/aws/resource_aws_iam_policy_attachment.go [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+		userErr = detachPolicyFromUsers(conn, users, arn)
+	}
+	if roles != nil {
+		roleErr = detachPolicyFromRoles(conn, roles, arn)
+	}
+	if groups != nil {
+		groupErr = detachPolicyFromGroups(conn, groups, arn)
+	}
+	if userErr != nil || roleErr != nil || groupErr != nil {
+		return fmt.Errorf("[WARN] Error removing user, role, or group list from IAM Policy Detach (%s), error:\n users - %v\n roles - %v\n groups - %v", name, userErr, roleErr, groupErr)
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_iam_policy_attachment.go]: Can we try something like this? 

```go 
errMsg := fmt.Sprintf("[WARN] Error removing user, role, or group list from IAM Policy Detach (%s), error(s):", name)
errs := []error{userErr, roleErr, groupErr}
for _, e := range errs {
  if e!= nil {
    errMsg = errMsg + "\n– " + e.Error()
  }
}

if userErr != nil || roleErr != nil || groupErr != nil { 
  return fmt.Errorf(errMsg)
}
```

That will omit and errors that didn't happen, say if we only had a user error and no others. Just thinking out loud

**Comment by catsby** [comment]
Some nit-picks, otherwise seems good :+1:

**Comment by GrayCoder** [comment]
I believe I took care of all the concerns, but please let me know if anyone sees anything else :smiley_cat:

**Comment by catsby** [comment]
Thanks @GrayCoder , this looks good to me

**Pull request merged**